src/allocator/GBM.cpp

```diff
@@ -188,6 +192,14 @@ SDMABUFAttrs Aquamarine::CGBMBuffer::dmabuf() {
     return attrs;
 }
 
+std::tuple<uint8_t*, uint32_t, size_t> Aquamarine::CGBMBuffer::beginDataPtr(uint32_t flags) {
+    uint32_t dst_stride = 0;
+    if (!boBuffer)
+        boBuffer = gbm_bo_map(bo, 0, 0, attrs.size.x, attrs.size.y, flags, &dst_stride, &gboMapping);
```

Comment:
if we call map twice, does that leak?

Reply:
It doesn't leak and doesn't update the flags. I've mentioned that. It creates the mapping on the first call and then just returns it every time. Currently there is no need in changing the flags.

Reply:
yes then question 2 applies, does it update when we change the buffer with the gpu?

Reply:
With `GBM_BO_TRANSFER_READ` or `GBM_BO_TRANSFER_READ_WRITE` it should. But I didn't test it.

Reply:
why not just unmap in `endDataPtr`?